syntax = "proto3";

import "Pipeline.proto";

package IRacingTools.Models;

message TelemetryDataFile {
  enum Status {
    STATUS_CREATED = 0;
    STATUS_AVAILABLE = 5;
    STATUS_INVALID = 10;
    STATUS_ERROR = 20;
  }

  string id = 1;
  string alias = 2;
  Status status = 3;

  int64 created_at = 5;
  int64 updated_at = 6;

  string filename = 10;
  
  string error_details = 20;
  
  // Pipeline (i.e. TRACK_MAP)
  repeated Pipeline pipelines = 50;

};


